File: algorithms/2_medium/M/02997_minimum_number_of_operations_to_make_array_xor_equal_to_k/bit_manipulation.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, int k) {
        int nums_xor {0};

        for (const int& num : nums) {
            nums_xor ^= num;
        }

        return (
            __builtin_popcount(nums_xor - (nums_xor & k)) +
            __builtin_popcount(k - (nums_xor & k))
        );
    }
};

File: algorithms/2_medium/M/02997_minimum_number_of_operations_to_make_array_xor_equal_to_k/bit_manipulation.rs
impl Solution {
    pub fn min_operations(nums: Vec<i32>, k: i32) -> i32 {
        let nums_xor = nums.into_iter().fold(0_i32, |xor, num| xor ^ num);
        ((nums_xor - (nums_xor & k)).count_ones() + (k - (nums_xor & k)).count_ones()) as i32
    }
}
